hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -20,20 +38,30 @@
 from hypothesis.internal.conjecture.junkdrawer import IntList, uniform
 from hypothesis.internal.conjecture.utils import calc_label_from_name
 
+if TYPE_CHECKING:
+    from hypothesis.strategies import SearchStrategy
+if sys.version_info < (3, 11):
+    from typing_extensions import dataclass_transform
+else:
+    from typing import dataclass_transform
```

Comment:
How would you like to deal with `dataclass_transform`?
Without it, MyPy will complain because all of the attrs classes are untyped.
But Hypothesis does not currently depend on `typing_extensions`.
We could make it required for testing, but not for users:

```python
if TYPE_CHECKING:
    from typing_extensions import dataclass_transform
else:
    def dataclass_transform() -> Callable[[T], T]:
        def wrapper(cls: T) -> T:
            return T
    return wrapper
```

Reply:
I think "required for type-checking, not required for users" is the ideal situation.

That would mean listing it in `tools.in`, so that that the tests themselves also run without optional dependencies - otherwise it's all too plausible that we accidentally make it a hard dependency someday.